api/src/main/java/com/aldoborrero/tinder/api/Tinder.java

```diff
@@ -45,6 +54,163 @@ protected Tinder(Configuration configuration) {
+        public Tinder build ();
+    }
+
+    protected RestAdapter getRestAdapter() {
+        if (restAdapter == null) {
+            restAdapter = new RestAdapter.Builder()
+                    .setEndpoint(settings.getEndpoint())
+                    .setClient(new OkClient(TinderOkHttpFactory.create(settings.getAuthTokenInterceptor())))
+                    .setConverter(new GsonConverter(TinderGsonFactory.create()))
+                    .setRequestInterceptor(
+                            new RequestInterceptor() {
+                                @Override
+                                public void intercept(RequestFacade requestFacade) {
+                                    List<Header> headerList = settings.getExtraHeaders();
+                                    if (headerList != null) {
+                                        for (Header header : headerList) {
+                                            requestFacade.addHeader(header.getName(), header.getValue());
+                                        }
+                                    }
+                                }
+                            }
+                    )
+                    .setErrorHandler(new TinderErrorHandler())
+                    .setLog(TinderLog.parseToRetrofit(settings.getLog()))
+                    .setLogLevel(TinderLogLevel.parseToRetrofit(settings.getLogLevel()))
+                    .build();
+        }
+        return restAdapter;
+    }
+
+    protected RestAdapter getAuthAdapter() {
+
+        if (authRestAdapter == null) {
+            authRestAdapter = new RestAdapter.Builder()
+                    .setEndpoint(settings.getEndpoint())
+                    .setClient(new OkClient(TinderOkHttpFactory.create()))
+                    .setConverter(new GsonConverter(TinderGsonFactory.create()))
+                    .setRequestInterceptor(new RequestInterceptor() {
+                        @Override
+                        public void intercept(RequestFacade requestFacade) {
+                            List<Header> headerList = settings.getExtraHeaders();
+                            if (headerList != null) {
+                                for (Header header : headerList) {
+                                    requestFacade.addHeader(header.getName(), header.getValue());
+                                }
+                            }
+                        }
+                    })
+                    .setErrorHandler(new TinderErrorHandler())
+                    .setLog(new RestAdapter.Log() {
+                        @Override
+                        public void log(String message) {
+                            // It's only for debug
+                            System.out.println(message);
+                        }
+                    })
+                    .setLogLevel(TinderLogLevel.parseToRetrofit(settings.getLogLevel()))
+                    .build();
+        }
+        return authRestAdapter;
+    }
+
+/*
```

Comment:
If this is commented because it's not needed it's better to remove it from the commit.